syntax = "proto3";

option go_package = "./signalling";

message Empty {
}

message PeerConfig {
    string peerConnectionOptions = 1;
}

message PlayerConnected {
    string playerId = 1;
}

service Signalling {
    rpc Connect(Empty) returns (Empty) {}
    rpc Config(Empty) returns (PeerConfig) {}

    // Viewer
    
    // Application
    rpc SubscribeToPlayerConnected(Empty) returns (stream PlayerConnected) {}
}